core/src/main/java/org/apache/camel/kafkaconnector/utils/CamelKafkaConnectDataformat.java

```diff
@@ -35,6 +80,147 @@ public CamelKafkaConnectDataformatKind getDataformatKind() {
 
     public enum CamelKafkaConnectDataformatKind {
         MARSHALL,
-        UNMARSHALL
+        UNMARSHALL;
+    }
+
+    public static void configureMarshalDataformat(RouteDefinition rd, CamelContext context, String dataFormatName, Properties camelProperties) {
+        Predicate<String> keyFilter = key -> key.contains(".marshal." + dataFormatName + ".");
+        DataFormatDefinition dfd = configureDataformat(context, dataFormatName, keyFilter, camelProperties);
+        if (dfd != null) {
+            rd.marshal(dfd);
+        } else {
+            rd.marshal(dataFormatName);
+        }
+    }
+
+
+    public static void configureUnmarshalDataformat(RouteDefinition rd, CamelContext context, String dataFormatName, Properties camelProperties) {
+        Predicate<String> keyFilter = key -> key.contains(".unmarshal." + dataFormatName + ".");
+        DataFormatDefinition dfd = configureDataformat(context, dataFormatName, keyFilter, camelProperties);
+        if (dfd != null) {
+            rd.unmarshal(dfd);
+        } else {
+            rd.unmarshal(dataFormatName);
+        }
+    }
+
+    public static DataFormatDefinition configureDataformat(CamelContext context, String dataFormatName, Predicate<String> keyFilter, Properties camelProperties) {
+        Map<String, Object> dataFormatProps = camelProperties.stringPropertyNames()
+                .stream()
+                .filter(keyFilter)
+                .collect(Collectors.toMap(key -> key.substring(key.lastIndexOf(".") + 1), key -> camelProperties.get(key)));
+
+        DataFormatDefinition dataformat;
+
+        switch (dataFormatName) {
```

Comment:
I hope this switch case could be avoided by something on the line of `context.createDataFormat(dataFormatName)`